### DISCUSSION TITLE
Slow XML and JSON Serialization

### DISCUSSION THREAD

**Discussion by a246530** [FirelyTeam/firely-net-sdk#2241] [Q&A]
I have a fairly large bundle that needs to be serialized for output. I noticed that this takes a really long time and the amount of memory used to complete the serialization process is insane. Think GB for what ends up being a 15-20MB return. Does anyone else have similar problems or has anyone found a way to reduce memory and time for serialization?

**Comment by ewoutkramer** [comment]
We have written a new set of parsers and serializers that are factors better in memory consumption than those that are currently in use in the SDK. They are quite new, so they are not being used in e.g. the FhirClient and validator yet.  But if you just want to parse data into POCO's, you can certainly check them out, they are documented here: https://docs.fire.ly/projects/Firely-NET-SDK/parsing/system-text-json-serialization.html, https://docs.fire.ly/projects/Firely-NET-SDK/parsing/system-text-json-deserialization.html (and there are equivalent XML versions).

On the longer term these will replace the current set of parsers for most scenarios.
